[ZhiWei1010] iP

ZhiWei1010 opened the pull request:
# Homie

**Homie** is your trusted personal assistant that helps you keep track of your tasks, events and deadlines.

## Choose **Homie** Because he's:
- Text-based ⌨️ 
- Easy to learn 📖 
- ~~fast~~ **SUPER FAST** 🔥 
- YOUR **HOMIE!!!!** 😎 

## Get Homie for **FREE** By following these steps:
1. Download _duke.jar_ from [here](https://github.com/ZhiWei1010/ip/releases)
2. Double-click it.
3. Add your tasks
4. Let **Homie** take care of the rest for you!

## Features
- [x] Add tasks (todo, events and deadline)
- [x] Delete tasks
- [x] List tasks
- [x] Mark/Unmark tasks as done/not done
- [x] Find tasks (using keyword search)
- [ ] Tasks, Events and Deadline Reminder (Coming Soon!)
- [ ] Import/Export Tasks (Coming Soon!)

> [!TIP]
> Useful for programmers/developers
## For Programmers/Developers
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(Duke.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

ZhiWei1010 committed:
Level 0. Renamed Duke to Homie, Greet and Exit.

ZhiWei1010 committed:
Level 1. Echo

ZhiWei1010 committed:
Level 2. Add, List, Changed the horizontal line to a variable so its easier to reuse.

ZhiWei1010 committed:
Level 3. Mark As Done
Added new Task class, Overriden the toString to display description of task.

ZhiWei1010 committed:
Level 4. ToDos, Events, Deadlines
Added Todo, Event and Deadline classes.

ZhiWei1010 committed:
A-TextUiTesting
Edited the Expected text and input text files.
removed a newline after horizontal line when using list command

ZhiWei1010 committed:
Level 5. Handle Errors
Added DukeException and TodoException classes.
implemented error catching of unrecognised commands and empty todo

ZhiWei1010 committed:
Level 6.
Added delete function

ZhiWei1010 committed:
Add duke package. Move all class to duke package.

ZhiWei1010 committed:
Start with feature to save, not complete yet.

ZhiWei1010 committed:
Add Save Function to Homie.
Add new Storage class.

ZhiWei1010 committed:
Level 8. Date and Times
Deadline: by variable is now of time LocalDateTime.
Format of Date accepted is "dd MM yyyy HHmm"
Format of Date printed is "MM-dd-yyyy HH:mm"

ZhiWei1010 committed:
Merge branch 'branch-Level-8'

wxiaoyun commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# Duke project template
+# duke.Duke project template
```
I love how when you refactored the duke class, you updated the readme files accordingly

ZhiWei1010 replied:
Thanks!

wxiaoyun commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,162 @@
+                        myList.add(currEvent);
+                        System.out.println("Got it. I've added this task:");
+                        System.out.println(currEvent);
+                        System.out.println("Now you have " + myList.size() + " tasks in the list.");
+                        System.out.println(line);
+                        storageFile.updateStorageFile(myList);
+                        break;
+                }
+                command = scanner.nextLine(); // Read next command
+                continue;
+            }
+
+            if (command.contains("mark")) {
+                tempArr = command.split(" ");
+                currTask = myList.get(Integer.parseInt(tempArr[1]) - 1);
+                switch (tempArr[0]) {
+                    case ("mark"):
+                        System.out.println(line);
+                        currTask.setDone();
+                        System.out.println("Nice! I've marked this task as done:");
+                        System.out.println(currTask);
+                        System.out.println(line);
+                        storageFile.updateStorageFile(myList);
+                        break;
+
+                    case ("unmark"):
+                        System.out.println(line);
+                        currTask.setNotDone();
+                        System.out.println("Ok, I've marked this task as not done yet:");
+                        System.out.println(currTask);
+                        System.out.println(line);
+                        storageFile.updateStorageFile(myList);
+                        break;
+
+                    default:
+                        System.out.println("Wrong Command");
+                }
+                command = scanner.nextLine(); // Read next command
+                continue;
+            }
```
Perhaps we can refactor the command interpretation into the Command class as a static method?

wxiaoyun commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,10 @@
+    public DukeException(String message) {
+        super(message);
+    }
+
+
+
```
Perhaps we can remove some unneeded new lines

wxiaoyun commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,86 @@
+6.[E][ ] SGO volunteering training  (from: Saturday 9am  to: 530pm)
+7.[E][ ] work at UCC  (from: Saturday 6pm  to: 10pm)
+____________________________________________________________
+____________________________________________________________
+Nice! I've marked this task as done:
+[T][X] Watch CS2106 Lecture
+____________________________________________________________
+____________________________________________________________
+Nice! I've marked this task as done:
+[D][X] Do CS2103 Quiz  (by: Friday 2359)
+____________________________________________________________
+____________________________________________________________
+Nice! I've marked this task as done:
+[E][X] coldplay concert  (from: Friday 8pm  to: 1030pm)
+____________________________________________________________
+____________________________________________________________
+Here are the tasks in your list:
+1.[T][X] Watch CS2106 Lecture
+2.[T][ ] Watch CS2107 Lecture
+3.[D][X] Do CS2103 Quiz  (by: Friday 2359)
+4.[D][ ] Do CS2103 IP  (by: Thursday 2359)
+5.[E][X] coldplay concert  (from: Friday 8pm  to: 1030pm)
+6.[E][ ] SGO volunteering training  (from: Saturday 9am  to: 530pm)
+7.[E][ ] work at UCC  (from: Saturday 6pm  to: 10pm)
+____________________________________________________________
+____________________________________________________________
+Ok, I've marked this task as not done yet:
+[T][ ] Watch CS2106 Lecture
+____________________________________________________________
+____________________________________________________________
+Ok, I've marked this task as not done yet:
+[D][ ] Do CS2103 Quiz  (by: Friday 2359)
+____________________________________________________________
+____________________________________________________________
+Ok, I've marked this task as not done yet:
+[E][ ] coldplay concert  (from: Friday 8pm  to: 1030pm)
+____________________________________________________________
+____________________________________________________________
+Bye. Hope to see you again soon!
+____________________________________________________________
```
Its great that you use tests to enforce the expected behavior of your app!

ZhiWei1010 replied:
Thanks!

wxiaoyun reviewed (approved):
LGTM. Your app hits all the specifications with some minor code standard nit picks

ZhiWei1010 committed:
Add TaskList class, Ui class, Parser Class.
Parser Class not in use yet.

ZhiWei1010 committed:
Merge branch 'A-MoreOOP'

ZhiWei1010 committed:
Merge branch 'master' into add-gradle-support

ZhiWei1010 committed:
Commit to undo Merge

ZhiWei1010 committed:
Revert "Commit to undo Merge"

This reverts commit c1c1c3b450571fd760cda5c244bdd2d7c1c157a5.

ZhiWei1010 committed:
Merge branch 'add-gradle-support' of https://github.com/ZhiWei1010/ip into add-gradle-support

ZhiWei1010 committed:
Merge branch 'add-gradle-support'
Merge add-gradle-support to master

ZhiWei1010 committed:
Gradle Set Up
Junit Set up

ZhiWei1010 committed:
Add JUnit Test for TaskList class and Todo

ZhiWei1010 committed:
Merge tag 'A-JUnit'
Create Tag for A-MoreOOP, A-Packages as well

ZhiWei1010 committed:
Add find function in TaskList

ZhiWei1010 committed:
Merge tag 'Level-9'

ZhiWei1010 committed:
Update JavaDoc for Half the classes

ZhiWei1010 committed:
Merge tag 'A-JavaDoc'

ZhiWei1010 committed:
Edit code to match coding standard.
Only violations are some missing javadocs

ZhiWei1010 committed:
Merge tag 'A-CodingStandard'

ZhiWei1010 committed:
Update build.gradle to create jar file with gradle

ZhiWei1010 committed:
Merge tag 'A-Jar'

chenyixin0 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+ */
+public class Deadline extends Task {
+
+    protected LocalDateTime by;
```
Maybe consider renaming this field (e.g. dueDate or completeBy) for greater clarity?

chenyixin0 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,73 @@
+
+    }
+
+    public void showWelcome() {
```
To standardise the method names in this class, considering changing this to showWelcomeMessage()?

ZhiWei1010 replied:
Thanks for spotting that! Will update this

chenyixin0 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,73 @@
+        System.out.println("What can I do for you?\n" + DIVIDER);
+    }
+
+    public void showGoodbye() {
```
Similar to above, can consider renaming this to showGoodbyeMessage()?

ZhiWei1010 replied:
Thanks for spotting that as well! Will update this!

chenyixin0 reviewed (commented):
Looks good to me! Good adherence to Java coding standards. Nice work!

ZhiWei1010 committed:
Add GUI to Homie chat bot.
Update missing JavaDocs

ZhiWei1010 committed:
Merge branch 'Leve-10'

ZhiWei1010 committed:
Add assertions in Duke

No assertions used before.

Add assertions to ensure that the inputs have the correct number of
inputs.

ZhiWei1010 committed:
Merge pull request #2 from ZhiWei1010/A-Assertions

Add assertions in Duke

ZhiWei1010 committed:
Merge branch 'master' of https://github.com/ZhiWei1010/ip

ZhiWei1010 committed:
Update Duke class

Update long methods in Duke.
Duke:run() and Duke: getResponse(String input) shortened by using
Parse class.

Update Parse class with processing methods for each command.

Update JavaDocs for Ui class, Duke class and Parse class.

ZhiWei1010 committed:
Merge pull request #3 from ZhiWei1010/A-CodeQuality

Update Duke class

ZhiWei1010 committed:
Merge branch 'master' of https://github.com/ZhiWei1010/ip

ZhiWei1010 committed:
Set up CI with GitHub Actions

Create ~/.github/workflows directory and add gradle.yml file to the
directory.

ZhiWei1010 committed:
Merge tag 'A-CI'

ZhiWei1010 committed:
Add feature C-FriendlierSyntax

User now can use aliases for commands.
Alias created:
* bye -> b
* list -> l
* mark -> m
* unmark -> unm
* delete -> del
* find -> f
* todo -> t
* deadline -> dead
* event -> e

ZhiWei1010 committed:
Merge tag 'BCD-Extension'

ZhiWei1010 committed:
Fix runtest.sh

Edit path to find the java files so CI can work

ZhiWei1010 committed:
Add print lines in Parser

Add print lines so chatbot will work in CLI and GUI

ZhiWei1010 committed:
Update Expected Text file for error testing

ZhiWei1010 committed:
Update input text for error testing

ZhiWei1010 committed:
Fix GUI unable to see more than 3 lines of text

Remove fixed height from Vbox.

ZhiWei1010 committed:
Fix Storage Class

Update storage class. Move static variables FILE_PATH and
CURRENT_DIRECTORY from Duke to Storage class.

ZhiWei1010 committed:
Update load task method in Storage class

loadTaskFromFile in Storage class method now prints 'No tasks in file.'
instead of 'Error loading tasks!'.

ZhiWei1010 committed:
Merge branch 'Fix-Storage'

ZhiWei1010 committed:
Refactor Duke to Homie

Rename all classes, methods and variables related to duke to homie.

ZhiWei1010 committed:
Merge branch 'Rename-Duke'

ZhiWei1010 committed:
Fix empty description for todo error message not showing

ZhiWei1010 committed:
Remove text-ui-test folder

All documents related to text-ui-test is removed as app does not use
it anymore.

ZhiWei1010 committed:
Update GUI to show welcome message

GUI does not show welcome message before.

GUI shows welcome message now when app starts up.

ZhiWei1010 committed:
Update exception handling

Exception thrown and catch in several levels before.

Exception thrown and catch in the highest level possible, so code is
consistent.

Remove all print statements and methods in Parser class.

Update TodoException message.

ZhiWei1010 committed:
Merge branch 'Fix-Bugs'

ZhiWei1010 committed:
Update User guide

ZhiWei1010 committed:
Update User Guide

ZhiWei1010 committed:
Update User Guide

ZhiWei1010 committed:
Update User guide

ZhiWei1010 committed:
Update user guide

ZhiWei1010 committed:
Fix todo description no spacing

ZhiWei1010 committed:
Improve Code Quality

Create new exception classes to handle relevant specific exceptions.

Update parser class to throw exceptions.

Update Ui class to format strings only instead of doing unrelated tasks.

ZhiWei1010 committed:
Fix storage bug

Update Deadline and Event class to fix bug in tasks not updating in
storage file.

ZhiWei1010 committed:
Update user guide

ZhiWei1010 committed:
Update Homie and user picture

ZhiWei1010 committed:
Upload Ui Screenshot

ZhiWei1010 committed:
Update Product Screenshot in README

ZhiWei1010 committed:
Merge branch 'A-UserGuide'

ZhiWei1010 committed:
Update Junit Test

Clean up format in output response and exception message

ZhiWei1010 committed:
Merge branch 'Improve-code'

ZhiWei1010 committed:
Update gradle.yml file

Remove IO redirection test as they are not needed anymore.

ZhiWei1010 committed:
Add Exit Application to README

ZhiWei1010 committed:
Update DialogBox and MainWindow.fxml to display more text in GUI

ZhiWei1010 committed:
Fix bugs in Parser

Parser able detect alias for deadline, event, todo commands now.

Fix find Command.

Add more checks in process commands to throw relevant exceptions instead
of generic HomieException.

ZhiWei1010 committed:
Update JavaDocs for all files

ZhiWei1010 committed:
Add assert statements in Storage

ZhiWei1010 committed:
Add isExit to Homie to check if command is exit command

Update MainWindow to close when exit command is detected.

Delete isExit from Parser.

ZhiWei1010 committed:
Update ParserTest and TaskListTest

ZhiWei1010 committed:
Merge branch 'Fix-Some-Bugs'
